pkg/ipcache/metadata.go

```diff
@@ -41,6 +41,11 @@ var (
 	// policy and propagated in monitor output.
 	identityMetadata = make(map[string]labels.Labels)
 
+	// applyIDMDChangesMutex protects InjectLabels and RemoveLabelsExcluded from being run in parallel
+	//
+	// This mutex is a temporary solution until a better approach is implemented
```

Comment:
nit: I don't think it's necessary to mention this. The solution is fine; it's not optimal but also unless we have immediate plans to implement a better approach, I don't think it really helps readers to reason about why this mutex exists.
```suggestion
```

Reply:
I removed this comment and force-pushed